runbotics-orchestrator/src/main/java/com/runbotics/web/rest/ProcessCollectionResource.java

```diff
@@ -36,28 +38,57 @@ public class ProcessCollectionResource {
     private final ProcessCollectionQueryService processCollectionQueryService;
 
     private final ProcessCollectionRepository processCollectionRepository;
+
+    private final UserService userService;
+
     @Value("${jhipster.clientApp.name}")
     private String applicationName;
 
     public ProcessCollectionResource(
         ProcessCollectionService processCollectionService,
         ProcessCollectionQueryService processCollectionQueryService,
-        ProcessCollectionRepository processCollectionRepository
+        ProcessCollectionRepository processCollectionRepository,
+        UserService userService
     ) {
         this.processCollectionService = processCollectionService;
         this.processCollectionRepository = processCollectionRepository;
         this.processCollectionQueryService = processCollectionQueryService;
+        this.userService = userService;
+    }
+
+    @PreAuthorize("@securityService.checkFeatureKeyAccess('" + FeatureKeyConstants.PROCESS_COLLECTION_READ + "')")
+    @GetMapping("process-collection/access")
+    public ResponseEntity getCollectionAccess(ProcessCollectionCriteria criteria) {
+        User requester = userService.getUserWithAuthorities().get();
+
+        if (criteria.getParentId() != null) {
+            processCollectionService.checkCollectionAvailability(
+                criteria.getParentId().getEquals(), requester
+            );
+        }
+
+        return ResponseEntity.accepted().build();
```

Comment:
does it return boolean?

Reply:
No, it just returns response code with no body. I also forget to pass void in ResponseEntity return type.

Reply:
let's add a comment above that tells that this endpoint returns 200 if user has access to particular collection and 403 if he/she is not. there should be examples in other .*Resource.java files